symengine/derivative.cpp

```diff
@@ -324,6 +323,12 @@ class DiffImplementation
         return fdiff(self, x);
     }
 
+    static RCP<const Basic> diff(const Dirichlet_eta &self,
```

Comment:
@isuruf `Dirichlet_eta` is a `OneArgFunction` and automatic differentiation via the `fdiff` routine is enabled by default only for `TwoArgFunction`.